client/my-sites/stats/stats-purchase/stats-pwyw-uprade-slider/index.tsx

```diff
@@ -0,0 +1,133 @@
+import formatCurrency from '@automattic/format-currency';
+import { useTranslate } from 'i18n-calypso';
+import TierUpgradeSlider from 'calypso/my-sites/stats/stats-purchase/tier-upgrade-slider';
+import { StatsPWYWSliderSettings } from 'calypso/my-sites/stats/stats-purchase/types';
+import './styles.scss';
+
+// TODO: Remove test data.
+// Currently used as a fallback if no plan info is provided.
+// Better approach is to require plan info and do some form of validation on it.
+function getPWYWPlanTiers( minPrice: number, stepPrice: number ) {
+	// From $0 to $20, in $1 increments.
+	let tiers: any[] = [];
+	for ( let i = 0; i <= 28; i++ ) {
+		tiers.push( {
+			price: formatCurrency( ( minPrice + i * stepPrice ) / 100, 'USD' ),
+			raw: minPrice + i * stepPrice,
+		} );
+	}
+	tiers = tiers.map( ( tier ) => {
+		const emoji = tier.raw < 500 ? ':|' : ':)';
+		return {
+			...tier,
+			lhValue: tier.price,
+			rhValue: emoji,
+		};
+	} );
+	return tiers;
+}
+
+function useTranslatedStrings() {
```

Comment:
I'm not saying this is bad, and I understand this would reduce noises in the component; however I find it handy when they are inlined - they kinda explains what the components are for and hence make it easier to understand the context.

And easier to search and locate 😄

Reply:
Noted! That's two thumbs down for this pattern so I'll bring the strings inline. 👍